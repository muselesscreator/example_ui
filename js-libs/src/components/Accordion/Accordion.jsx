/** @module */
import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { Accordion as BaseAccordion } from 'semantic-ui-react';

import './Accordion.scss';

/**
 * @callback selectCallback
 * @param {Object} e - event object
 * @param {Object} - clicked title element data
 */

/**
 * Simple Accordion overlay that adds our theming
 * Accepts all options for the semantic-ui-react Accordion
 *
 * <https://react.semantic-ui.com/modules/accordion>
 *
 * @param {string=} className - additional css classes
 * @param {selectCallback=} onSelect - callback to called w/ -1 if the element
 *   is active, or the index of the selected title card otherwise.
*/
export const Accordion = ({ className, onSelect, ...props }) =>
  <BaseAccordion
    className={classNames('brw-accordion', className)}
    styled
    onTitleClick={(e, { index, active }) => onSelect( active ? -1 : index )}
    inverted
    {...props}
  />

Accordion.propTypes = {
  className: PropTypes.string,
  onSelect: PropTypes.func,
}
Accordion.defaultProps = {
  onSelect: () => ({}),
};

export default Accordion;
