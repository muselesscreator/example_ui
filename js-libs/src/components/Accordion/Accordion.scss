@import "../../scss/variables.scss";
.ui.styled.brw-accordion.accordion, 
.ui.styled.brw-accordion.accordion .accordion {
  >.title, >.active.title {
    color: white !important;
  }
  background-color: $panel-main;
  border: 1px groove $border-main;
  >.title {
    font-size: 20px;
    border-bottom: 1px solid white;
    border-top: 1px solid white;
    background-color: $panel-med-dark;
    &.active {
      background-color: $panel-med-light;
    }
  }
  +.label {
    margin-bottom: 20px;
    padding: 10px;
    background-color: #484d59 !important;
    border: 1px solid #595d66 !important;
    color: white !important;
  }
}
