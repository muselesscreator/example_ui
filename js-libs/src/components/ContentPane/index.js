import ContentPane from './ContentPane';

export default ContentPane;
