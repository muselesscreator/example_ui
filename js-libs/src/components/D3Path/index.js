import D3Path from './D3Path';

export default D3Path;
