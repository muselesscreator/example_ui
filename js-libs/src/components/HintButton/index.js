import HintButton from './HintButton';

export default HintButton;
