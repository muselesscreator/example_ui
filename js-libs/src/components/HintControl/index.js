import HintControl from './HintControl';

export default HintControl;
