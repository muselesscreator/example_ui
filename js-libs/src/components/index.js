export { default as Accordion } from './Accordion';
export { default as Button } from './Button';
export { default as Canvas } from './Canvas';
export { default as Checkbox } from './Checkbox';
export { default as Component } from './Component';
export { default as ContentColumn } from './ContentColumn';
export { default as ContentPane } from './ContentPane';
export { default as D3Path } from './D3Path';
export { default as Divider } from './Divider';
export { default as Dropzone } from './Dropzone';
export { default as FormErrors } from './FormErrors';
export { default as HintButton } from './HintButton';
export { default as HintControl } from './HintControl';
export { default as HintLabel } from './HintLabel';
export { default as Input } from './Input';
export { default as MenuButton } from './MenuButton';
export { default as ModalButton } from './ModalButton';
export { default as Nav } from './Nav';
export { default as Number } from './Number';
export { default as Select } from './Select';
export { default as Slider } from './Slider';
export { default as Table } from './Table';
export { default as ValueLabel } from './ValueLabel';
