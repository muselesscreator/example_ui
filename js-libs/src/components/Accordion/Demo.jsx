import React from 'react';

import Accordion from './Accordion';

const LabelComponent = ({ value }) => <div>{value}</div>;

export const Demo = ({ className, ...props }) => {
  const mkPanel = (title, content) => ({ key: title, title, content: { content } });
  return (
    <div>
      <Accordion
        className='test-example'
        panels={[
          { key: 1, title: 'Basic', content: { content: "An Entry" } },
          {
            key: 2,
            title: 'Component',
            content: { content: <LabelComponent value="Label Component" /> }
          },
          mkPanel('Short-hand', <LabelComponent value="Short-Hand Component" />),
        ]}
      />
    </div>
  );
}

export default Demo;
