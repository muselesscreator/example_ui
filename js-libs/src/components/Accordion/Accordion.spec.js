import _ from 'lodash';

import { render, simpleTest } from 'modules/redux-test/components';
import { Accordion } from './Accordion';

describe('Accordion component', () => {
  const props = { some: 'props', forThe: 'tests' };
  const className = 'cln';
  simpleTest(
    'basic',
    Accordion,
    {
      className,
      onSelect: jest.fn(),
      ...props,
    }
  );
  describe('prop tests', () => {
    let el, subElement, elementProps, onSelect;

    beforeEach(() => {
      onSelect = jest.fn();
      el = render(Accordion, {
        className,
        onSelect,
        ...props,
      });
      subElement = el.find('Accordion');
      elementProps = subElement.props();
    });

    it('forwards props', () => {
      expect(elementProps.className.split(' ')).toContain(className);
      _.mapValues(props, (val) => expect(elementProps[val]).toEqual(props[val]));
    });

    it('calls onSelect on title click (index if inactive, -1 if active)', () => {
      const index = 2;
      elementProps.onTitleClick('err', { index, active: true });
      expect(onSelect).toHaveBeenCalledWith(-1);
      elementProps.onTitleClick('err', { index, active: false });
      expect(onSelect).toHaveBeenCalledWith(index);
    });
  });

});
