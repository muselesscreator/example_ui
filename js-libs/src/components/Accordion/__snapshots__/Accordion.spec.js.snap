// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component :basic 1`] = `
<Accordion
  className="brw-accordion cln"
  forThe="tests"
  inverted={true}
  onTitleClick={[Function]}
  some="props"
  styled={true}
/>
`;
